### ISSUE TITLE
Private Videos

### ISSUE THREAD

**Issue by 19peaches** [dobromir-hristov/vue-vimeo-player#13]
Hey, I love your package, but I do need to be able to embed our private company videos, not just public ones. Is this possible with this package or will I need to look for another solution?

I looked through your code and I don't see a way to authenticate.

Thanks.

**Comment by dobromir-hristov** [comment]
I need to check how vimeo api does it and can add the featureOn 27 Sep 2018 08:44, Vince Kronlein <notifications@github.com> wrote:Hey, I love your package, but I do need to be able to embed our private company videos, not just public ones. Is this possible with this package or will I need to look for another solution?
I looked through your code and I don't see a way to authenticate.
Thanks.

**Comment by dobromir-hristov** [comment]
Could you tell me what error you are getting? I think this is an overall issue with vimeo/player.js, at least from what I saw on their GitHub repo.

**Comment by 19peaches** [comment]
I'm just getting a 404 nothing else.

**Comment by Chemahin** [comment]
Hi everyone !) I use your component but have a problem. prop autoplay doesn't work ((( what am I doing wrong ?
<template>
  <vimeo-player
    ref="player"
    :video-id="videoID"
    @ready="onReady"
    :autoplay="autoplay"
    :player-height="height"
    :player-width="width"
    :loop="true">
  </vimeo-player>
</template>
<script>
  export default {
    data() {
      return {
        autoplay: true,
        videoID: '1',
        height: 1000,
        width: 1300,
        options: {},
        playerReady: false
      }
    }
 thanks for the answer)

**Comment by dobromir-hristov** [comment]
If its on a mobile phone, then no, that wont work. 

Make an example on Codesandbox so I can check it out better.

**Comment by josephharveyangeles** [comment]
Hi @19peaches, how were you able to go about doing so? There's a whitelisting feature in vimeo. I'm assuming that vimeo will take care of everything. Just curious if you've encountered a certain problem.

**Comment by 19peaches** [comment]
@josephharveyangeles 

I wasn't able to. I switched to a different package and player.

**Comment by dobromir-hristov** [comment]
@19peaches are you still using vimeo? You can share a link to what you chose for future users.

**Comment by 19peaches** [comment]
@dobromir-hristov yes I'm still using Vimeo via the Plyr package here: https://github.com/sampotts/plyr

I'm not entirely happy with it, but it works pretty well.

**Comment by josephharveyangeles** [comment]
Also, won't the whitelisting work out of the box? Just add the domain on the whitelist. Oh, but yeah you won't be able to see it on localhost...

**Comment by 19peaches** [comment]
Whitelisting the domain did not work for me, hence the reason I switched players.

**Comment by dobromir-hristov** [comment]
Yes, they use a different player, not the vimeo official one. Sad vimeo dont fix their own stuff... This package uses the official player from Vimeo, so nothing I can do guys, sorry.

**Comment by josephharveyangeles** [comment]
@19peaches can you share what particular issue you encountered about the whitelisting? I just finished embedding videos on my app. Was about to upgrade to vimeo pro tomorrow.

**Comment by 19peaches** [comment]
@josephharveyangeles 

I already answered this earlier, I received a 404 not found when attempting to embed a private video. This includes domain protected videos. Nothing further to add really.

**Comment by mesaglam** [comment]
@19peaches  If you want to play domain protected videos at outside, you have to build a proxy script in your allowed domain. 

You can build a player page in your domain; then you may call that page inside an iframe from wherever you want...

There is no way else...

**Comment by DiAvisoo** [comment]
The problem is that unlisted videos have an additional STRING element in the identifier:

"NOTE: When you want to embed a video with the unlisted privacy setting, be sure to send the entire URL, not just the portion with the video ID. The URL of an unlisted video contains an additional string of characters that you must include with your GET request to receive embed data of any kind."

**Comment by CHEWX** [comment]
I'm missing the issue here, you just add your local domain to the domain exclusion list? Definitely not an issue with this package as I'm using this exact setup with no issue, if I remove my local domain, then the video 404's as you've mentioned. 

Try a proxy.

**Comment by seguce92** [comment]
@19peaches  I managed to use the component and play private videos using the api, something like that
`<vimeo-player
      ref="player"
      :options="options"
      video-url="https://player.vimeo.com/video/{video_id}"
      :video-id="null"
      :autoplay="false"
      :player-height="320"
      :player-width="640">
</vimeo-player>`

You get the url by clicking on the video > insert code.

**Comment by pongstylin** [comment]
@dobromir-hristov 
There's a bug in the implementation of it, it appears.  The Vimeo API expects a video ID **or** a video URL.  Not both.  When I pass both, I get a 404 error.  When I only pass the URL, it works - but I get a nasty warning in the console saying that the ID is required.  Either don't make the ID property required, or don't pass the ID to the vimeo player when passing the URL.

**Comment by dobromir-hristov** [comment]
Great find, care to make a PR? I am not using the package anymore so such things might happen.

**Comment by pongstylin** [comment]
I don't think I can put the level of TCL that would be required with the change.  I see that you have branched for vue2 and vue3 and the change would need to be applied and tested in both.  But I'm just working with vue2 right now.  Also, the recent pull request that fixes the "update" method to support options needs to be taken into consideration as well to ensure both ID and URL based configuration works appropriately.

**Comment by rangermeier** [comment]
For reference:
This has been fixed in [v1.0.4](https://github.com/dobromir-hristov/vue-vimeo-player/releases/tag/v1.0.4).
It's now possible to pass only a `video-url` in the props which works with unlisted (private) videos:

```
<vimeo-player video-url="https://vimeo.com/video/{video_id}/{hash}" />
```
